src/main/java/teammates/ui/controller/XlsxFileDownloadResult.java

```diff
@@ -0,0 +1,102 @@
+package teammates.ui.controller;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.poi.ss.usermodel.IndexedColors;
+import org.apache.poi.xssf.usermodel.XSSFCell;
+import org.apache.poi.xssf.usermodel.XSSFCellStyle;
+import org.apache.poi.xssf.usermodel.XSSFFont;
+import org.apache.poi.xssf.usermodel.XSSFRow;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+import com.google.common.base.Strings;
+
+import teammates.common.datatransfer.AccountAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.StatusMessage;
+
+public class XlsxFileDownloadResult extends FileDownloadResult {
+    public static final String CSV_LINE_DELIMITTER_REGEX = ",(?=([^\"]*\"[^\"]*\")*[^\"]*$)";
+    public static final String QUESTION_LINE_REGEX = "Question [0-9]{1,}.{0,}";
+    private XSSFWorkbook workBook;
+
+    public XlsxFileDownloadResult(String destination, AccountAttributes account, List<StatusMessage> status,
+            String fileName, String fileContent, String fileType) {
+        super(destination, account, status, fileName, fileContent, fileType);
+        createWorkBook();
+    }
+
+    public XSSFWorkbook getWorkBook() {
+        return workBook;
+    }
+
+    @Override
+    public void send(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        resp.setContentType("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");
+        resp.setHeader("Content-Disposition", getContentDispositionHeader());
+        ServletOutputStream outStream = resp.getOutputStream();
+        workBook.write(outStream);
+        outStream.flush();
+        outStream.close();
+    }
+
+    public XSSFWorkbook createWorkBook() {
+        workBook = new XSSFWorkbook();
+        XSSFSheet sheet = workBook.createSheet("sheet1");
+        int rowNumber = 0;
+        if (!Strings.isNullOrEmpty(fileContent)) {
+            String[] lines = fileContent.split(Const.EOL);
+            for (String line : lines) {
+                // Split by comma only if comma has a zero or even number of quotes infront.
+                String[] str = line.split(CSV_LINE_DELIMITTER_REGEX, -1);
```

Comment:
Hmm, this still seems a little fragile. If the question or response text contains quotations and commas, then it is likely that this approach will fail. I suspect we have to make changes at the code before `FileDownloadResult` is used. Instead of constructing csv formatted file contents then trying to convert the csv formatted contents into xls, we should try to construct the data in a different way, or construct the `XSSFWorkbook` in the `*Logic` classes.